### ISSUE TITLE
Web-based file manager - Root path not found

### ISSUE THREAD

**Issue by cpu150** [GeekFunkLabs/squishbox#7]
# Description
After a fresh install and reboot I got an error while trying to access the web-based file manager.
After logging in this error message is shown: `Root path "/home/cpu150/SquishBox" not found!`

<img width="1481" alt="Web 404" src="https://github.com/GeekFunkLabs/squishbox/assets/13674265/35da6e3a-7a8c-4474-a7d7-184ff079a694">

# Steps to reproduce
1. Flash `Raspberry Pi Os Lite (64-bit)` (Debian Bookworm - no desktop environment) image
2. Update system: `sudo apt update ; sudo apt -y upgrade`
3. Install SquishBox as `Naked Raspberry Pi Synth`
4. Reboot
5. Access the Web-based file manager from a web browser
6. Log in

Result: Get a not found error message
Expected: Should display the Squishbox root folder content

# Setup
Raspberry Pi Zero 2W:
- Operating System: Debian GNU/Linux 12 (bookworm)
- Kernel: Linux 6.6.28+rpt-rpi-v8
- Architecture: arm64

Squishbox:
- Naked Raspberry Pi Synth
- USB audio output - card 0: Audio [Hi-Res Audio], device 0: USB Audio [USB Audio]

**Comment by cpu150** [comment]
It seems to be a rights issue:
My home directory is set as: `drwx------  4 cpu150 cpu150 4096 May 16 15:24 cpu150`
If I change as `chmod 750 ~`:  `drwxr-x---  4 cpu150 cpu150 4096 May 16 15:24 cpu150`
then it works fine.
Which makes sense `nginx` server is ran by `www-data` user.

The above allows us to confirm where the issue comes from however I would not recommend that as a solution as it introduces security issues.
I don't have a proper solution for that.
Maybe changing the default installation folder?

**Comment by albedozero** [comment]
Debian 12 Bookworm (or at least the Raspberry Pi OS version of it) causes this problem by using default permissions 700 (`drwx------`) for home directories. Up until Bullseye it was 755 (`drwxr-xr-x`), which works fine as long as `www-data` is added to the user's group.

I think the solution is to make sure directories in the path to `SquishBox/`, wherever it is installed, are group-executable (i.e. `drwx--x---`). That way `www-data` can traverse them but not write anywhere it isn't supposed to. Alternate default locations might be `/usr/local/share/SquishBox` or `/srv/SquishBox`, but those will require permission/ownership changes as well, and I feel like the home directory is the easiest place for users to find bank and config files if they log into the Pi.

**Comment by cpu150** [comment]
Indeed: I just tried to give just the executable right and it sounds working fine (`drwx--x---`).
I got your point and do agree: it's easier to find if in the home directory.
Maybe a symbolic link could be a work around?

Also (but this connects with another big topic so happy to have it somewhere else than here), I have seen another project with the system in read only which makes it easier to shutdown (no way to corrupt a file so no need a command to power it off).
If this sounds interesting to you then I would suggest to create another partition (rw) and install `SquishBox` files there while keeping the system partition as read only.
